components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1197,7 +1205,28 @@ public IdentityProvider getIdPByName(Connection dbConnection, String idPName, in
     public IdentityProvider getIDPbyId(Connection dbConnection, int idpId, int tenantId,
                                        String tenantDomain) throws IdentityProviderManagementException {
 
-        return getIDP(dbConnection, null, idpId, tenantId, tenantDomain);
+        ExtendedIdentityProvider extendedIdp = getIDP(dbConnection, null, idpId, null, tenantId, tenantDomain);
+        if (extendedIdp != null) {
+            return extendedIdp.getIdentityProvider();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Retrieves an IDP by it's ID.
+     *
+     * @param dbConnection Database Connection.
+     * @param resourceId   Identity Provider Resource ID.
+     * @param tenantId     Tenant ID of the IDP.
+     * @param tenantDomain Tenant Domain of the IDP.
+     * @return An Identity Provider with given name.
+     * @throws IdentityProviderManagementException IdentityProviderManagementException
+     */
+    public ExtendedIdentityProvider getIDPbyResourceId(Connection dbConnection, String resourceId, int tenantId,
```

Comment:
Why would someone call this method with a database connection?

Reply:
in addIDP, updateIDP, this method is called with db connection to retrieve the created/ updated IDP object.